k8saudit-eks plugin crashes due to missing maxEventBytes configuration support

Hello all! 

The k8saudit-eks Falco plugin crashes on EKS v1.34 when receiving large Kubernetes audit log events.
The upstream [k8saudit plugin](https://github.com/falcosecurity/charts/blob/master/charts/falco/values-k8saudit.yaml#L51) supports maxEventBytes configuration to safely limit payload parsing — but k8saudit-eks does not include this parameter.
As a result, large payloads trigger a slice bounds out of range panic inside the JSON parser (fastjson):
```
panic: runtime error: slice bounds out of range [1636226:1048376]
```

Component Versions:
AWS EKS -1.34
Falco Helm Chart - 4.22.0
Falco appVersion - 0.41.0
Plugin - k8saudit-eks v0.16.0


Stack trace
```
Thu Nov 27 12:40:49 2025: [libs]: Trying to open the right engine!
panic: runtime error: slice bounds out of range [1636226:1048376]

goroutine 8 [running]:
github.com/valyala/fastjson.parseValue({0xc001394000, 0xfff38}, 0x9f7d3, 0xc00003fde8?, 0x7f08bb038a00?)
	/go/pkg/mod/github.com/geraldcombs/fastjson@v0.0.0-20250801170450-bf39244e60b8/parser.go:121 +0x1155
github.com/valyala/fastjson.parseObject({0xc001394000, 0xfff38}, 0x9f6fb, 0xc00003fde8, 0x3)
	/go/pkg/mod/github.com/geraldcombs/fastjson@v0.0.0-20250801170450-bf39244e60b8/parser.go:284 +0x692
github.com/valyala/fastjson.parseValue({0xc001394000, 0xfff38}, 0x9f6fb, 0xc00003fde8?, 0x7f08bb038a00?)
	/go/pkg/mod/github.com/geraldcombs/fastjson@v0.0.0-20250801170450-bf39244e60b8/parser.go:112 +0xfd7
github.com/valyala/fastjson.parseObject({0xc001394000, 0xfff38}, 0x9efc8, 0xc00003fde8, 0x2)
	/go/pkg/mod/github.com/geraldcombs/fastjson@v0.0.0-20250801170450-bf39244e60b8/parser.go:284 +0x692
github.com/valyala/fastjson.parseValue({0xc001394000, 0xfff38}, 0x9efc8, 0xc0001bdde8?, 0x7f08bb038a00?)
	/go/pkg/mod/github.com/geraldcombs/fastjson@v0.0.0-20250801170450-bf39244e60b8/parser.go:112 +0xfd7
github.com/valyala/fastjson.parseObject({0xc001394000, 0xfff38}, 0x0, 0xc00003fde8, 0x1)
	/go/pkg/mod/github.com/geraldcombs/fastjson@v0.0.0-20250801170450-bf39244e60b8/parser.go:284 +0x692
github.com/valyala/fastjson.parseValue({0xc001394000, 0xfff38}, 0x0, 0x2?, 0xc0004321c0?)
	/go/pkg/mod/github.com/geraldcombs/fastjson@v0.0.0-20250801170450-bf39244e60b8/parser.go:112 +0xfd7
github.com/valyala/fastjson.(*Parser).Parse(0xc0001bddd0, {0xc000f26000, 0xfff38})
	/go/pkg/mod/github.com/geraldcombs/fastjson@v0.0.0-20250801170450-bf39244e60b8/parser.go:36 +0x125
github.com/valyala/fastjson.(*Parser).ParseBytes(...)
	/go/pkg/mod/github.com/geraldcombs/fastjson@v0.0.0-20250801170450-bf39244e60b8/parser.go:53
github.com/valyala/fastjson.ParseBytes({0xc000f26000?, 0xc00066d701?, 0x2?})
	/go/pkg/mod/github.com/geraldcombs/fastjson@v0.0.0-20250801170450-bf39244e60b8/handy.go:157 +0x35
github.com/falcosecurity/plugins/plugins/k8saudit/pkg/k8saudit.(*Plugin).ParseAuditEventsPayload(0xc0004257a0, {0xc000f26000?, 0x0?, 0x0?})
	/go/pkg/mod/github.com/falcosecurity/plugins/plugins/k8saudit@v0.16.0/pkg/k8saudit/source.go:265 +0x26
github.com/falcosecurity/plugins/plugins/k8saudit-eks/pkg/k8sauditeks.(*Plugin).Open.func1()
	/__w/plugins/plugins/plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go:161 +0x2c7
created by github.com/falcosecurity/plugins/plugins/k8saudit-eks/pkg/k8sauditeks.(*Plugin).Open in goroutine 17
	/__w/plugins/plugins/plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go:147 +0x2d1
```

--- a/plugins/k8saudit-eks/README.md
+++ b/plugins/k8saudit-eks/README.md
@@ -129,6 +129,8 @@ load_plugins: [k8saudit-eks, json]
  * `polling_interval`: Polling Interval in seconds (default: 5s)
  * `shift`: Time shift in past in seconds (default: 1s)
  * `buffer_size`: Buffer Size (default: 200)
+ * `max_event_size`: Maximum size of single audit event (default: 262144)
+
 
 **Open Parameters**
 A string which contains the name of your EKS Cluster (required).

--- a/plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go
+++ b/plugins/k8saudit-eks/pkg/k8sauditeks/k8sauditeks.go
@@ -55,6 +55,7 @@ type PluginConfig struct {
 	Shift           uint64 `json:"shift"            jsonschema:"title=shift,description=Time shift in past in seconds (default: 1s),default=1"`
 	PollingInterval uint64 `json:"polling_interval" jsonschema:"title=polling_interval,description=Polling Interval in seconds (default: 5s),default=5"`
 	UseAsync        bool   `json:"use_async"        jsonschema:"title=use_async,description=If true then async extraction optimization is enabled (default: true),default=true"`
+	MaxEventSize    uint64 `json:"max_event_size"   jsonschema:"title=max_event_size,description=Maximum size of single audit event (default: 262144),default=262144"`
 }
 
 func (k *Plugin) Info() *plugins.Info {
@@ -83,6 +84,7 @@ func (p *PluginConfig) Reset() {
 		p.Region = i
 	}
 	p.UseAsync = true
+	p.MaxEventSize = uint64(sdk.DefaultEvtSize)
 	// for PollingInterval, Shift and BufferSize, the default values from the package are used automatically
 }
 
@@ -103,6 +105,9 @@ func (k *Plugin) Init(cfg string) error {
 
 	// setup optional async extraction optimization
 	extract.SetAsync(k.Config.UseAsync)
+
+	// configure base k8saudit plugin with MaxEventSize
+	k.Plugin.Config.MaxEventSize = k.Config.MaxEventSize
 
 	k.Logger = log.New(os.Stderr, "["+pluginName+"] ", log.LstdFlags|log.LUTC|log.Lmsgprefix)
 
@@ -180,5 +185,6 @@ func (p *Plugin) Open(clustername string) (source.Instance, error) {
 	return source.NewPushInstance(
 		pushEventC,
 		source.WithInstanceClose(cancel),
+		source.WithInstanceEventSize(uint32(p.Config.MaxEventSize)),
 	)
 }

